commands/http/handler.go

```diff
@@ -165,44 +186,32 @@ func (i internalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	streamChans, _, _ := req.Option("stream-channels").Bool()
-	if isChan && streamChans {
-		// w.WriteString(transferEncodingHeader + ": chunked\r\n")
-		// w.Header().Set(channelHeader, "1")
-		// w.WriteHeader(200)
-		err = copyChunks(applicationJson, w, out)
-		if err != nil {
-			log.Debug("copy chunks error: ", err)
+	if isChan {
+		h.Set(channelHeader, "1")
+		if streamChans {
+			// streaming output from a channel will always be json objects
+			mime = applicationJson
 		}
-		return
 	}
 
-	err = flushCopy(w, out)
-	if err != nil {
-		log.Debug("Flush copy returned an error: ", err)
+	if mime != "" {
+		h.Set(contentTypeHeader, mime)
 	}
-}
-
-func (i Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// Call the CORS handler which wraps the internal handler.
-	i.corsHandler.ServeHTTP(w, r)
-}
+	h.Set(streamHeader, "1")
+	h.Set(transferEncodingHeader, "chunked")
 
-// flushCopy Copies from an io.Reader to a http.ResponseWriter.
-// Flushes chunks over HTTP stream as they are read (if supported by transport).
-func flushCopy(w http.ResponseWriter, out io.Reader) error {
-	if _, ok := w.(http.Flusher); !ok {
-		return copyChunks("", w, out)
+	if err := copyChunks(status, w, out); err != nil {
+		log.Error("error while writing stream", err)
 	}
-
-	_, err := io.Copy(&flushResponse{w}, out)
-	return err
```

Comment:
It was because `io.Copy` writes to the buffered writer, but didnt flush it afterwards, so we made that flushResponse thing to make sure a flush happened after each write. We flush properly within copyChunks